examples/style.rs

```diff
@@ -0,0 +1,238 @@
+use std::{
+    error::Error,
+    io::{self, Stdout},
+    iter::once,
+    result,
+    time::Duration,
+};
+
+use crossterm::{
+    event::{self, Event, KeyCode},
+    execute,
+    terminal::{disable_raw_mode, enable_raw_mode, EnterAlternateScreen, LeaveAlternateScreen},
+};
+use ratatui::{prelude::*, widgets::*};
+
+type Result<T> = result::Result<T, Box<dyn Error>>;
+
+fn main() -> Result<()> {
+    let mut terminal = setup_terminal()?;
+    let res = run_app(&mut terminal);
+    restore_terminal(terminal)?;
+    if let Err(err) = res {
+        eprintln!("This example will fail terminal height less than 32 lines.");
+        eprintln!("{err:?}");
+    }
+    Ok(())
+}
+
+fn run_app<B: Backend>(terminal: &mut Terminal<B>) -> io::Result<()> {
+    loop {
+        terminal.draw(ui)?;
+
+        if event::poll(Duration::from_millis(250))? {
+            if let Event::Key(key) = event::read()? {
+                if let KeyCode::Char('q') = key.code {
+                    return Ok(());
+                }
+            }
+        }
+    }
+}
+
+fn ui<B: Backend>(frame: &mut Frame<B>) {
+    let normal_colors = [
+        Color::Black,
+        Color::Red,
+        Color::Green,
+        Color::Yellow,
+        Color::Blue,
+        Color::Magenta,
+        Color::Cyan,
+        Color::Gray,
+    ];
+    let bright_colors = [
+        Color::DarkGray,
+        Color::LightRed,
+        Color::LightGreen,
+        Color::LightYellow,
+        Color::LightBlue,
+        Color::LightMagenta,
+        Color::LightCyan,
+        Color::White,
+    ];
+    render_bg_colors(frame, normal_colors, bright_colors);
+    render_fg_colors(frame, normal_colors, bright_colors);
+    render_indexed_colors(frame);
+    render_indexed_grayscale(frame);
+    render_modifiers(frame);
+}
+
+fn render_bg_colors<B: Backend>(
+    frame: &mut Frame<B>,
+    normal_colors: [Color; 8],
+    bright_colors: [Color; 8],
+) {
+    for (i, color) in normal_colors.iter().enumerate() {
+        let paragraph = Paragraph::new(color.to_string()).style(Style::default().bg(*color));
+        frame.render_widget(paragraph, Rect::new(i as u16 * 8, 0, 8, 1));
+    }
+    for (i, color) in bright_colors.iter().enumerate() {
+        let paragraph = Paragraph::new(color.to_string()).style(Style::default().bg(*color));
+        frame.render_widget(paragraph, Rect::new(i as u16 * 8, 1, 8, 1));
+    }
+    for (i, color) in normal_colors.iter().enumerate() {
+        let paragraph =
+            Paragraph::new(color.to_string()).style(Style::default().bg(*color).fg(Color::Black));
+        frame.render_widget(paragraph, Rect::new(i as u16 * 8, 2, 8, 1));
+    }
+    for (i, color) in bright_colors.iter().enumerate() {
+        let paragraph =
+            Paragraph::new(color.to_string()).style(Style::default().bg(*color).fg(Color::Black));
+        frame.render_widget(paragraph, Rect::new(i as u16 * 8, 3, 8, 1));
+    }
+}
```

Comment:
What do you think about adding some kind of title for each row? It's easier to grok what is going on when looking at the screenshot that way. e.g.:

<img width="464" alt="image" src="https://github.com/ratatui-org/ratatui/assets/1813121/a7581638-35e1-41fb-b399-c95877eb804d">


```suggestion
fn render_bg_colors<B: Backend>(
    frame: &mut Frame<B>,
    normal_colors: [Color; 8],
    bright_colors: [Color; 8],
) {
    frame.render_widget(
        Paragraph::new("normal colors with white foreground"),
        Rect::new(0, 0, 50, 1),
    );
    for (i, color) in normal_colors.iter().enumerate() {
        let paragraph = Paragraph::new(color.to_string()).style(Style::default().bg(*color));
        frame.render_widget(paragraph, Rect::new(i as u16 * 8, 1, 8, 1));
    }
    frame.render_widget(
        Paragraph::new("bright colors with white foreground"),
        Rect::new(0, 2, 50, 1),
    );
    for (i, color) in bright_colors.iter().enumerate() {
        let paragraph = Paragraph::new(color.to_string()).style(Style::default().bg(*color));
        frame.render_widget(paragraph, Rect::new(i as u16 * 8, 3, 8, 1));
    }
    frame.render_widget(
        Paragraph::new("normal colors with black foreground"),
        Rect::new(0, 4, 50, 1),
    );
    for (i, color) in normal_colors.iter().enumerate() {
        let paragraph =
            Paragraph::new(color.to_string()).style(Style::default().bg(*color).fg(Color::Black));
        frame.render_widget(paragraph, Rect::new(i as u16 * 8, 5, 8, 1));
    }
    frame.render_widget(
        Paragraph::new("bright colors with black foreground"),
        Rect::new(0, 6, 50, 1),
    );
    for (i, color) in bright_colors.iter().enumerate() {
        let paragraph =
            Paragraph::new(color.to_string()).style(Style::default().bg(*color).fg(Color::Black));
        frame.render_widget(paragraph, Rect::new(i as u16 * 8, 7, 8, 1));
    }
}
```

Reply:
Sounds good

Reply:
The problem with doing this is that I'm already height constrained at least on the terminal settings I use on my Macbook 16in which hits 171x51, and width constrained as I want a consistent size for all the examples that fits well into the same width vhs dimensions (which ended up nicely hitting 80 chars).

Perhaps it's worth splitting colors and modifiers into two examples.

Reply:
I split the examples